plugins/utils/tmux-navigator.nix

```diff
@@ -6,37 +6,127 @@
   ...
 }:
 with lib; let
-  cfg = config.plugins.tmux-navigator;
-in {
-  options.plugins.tmux-navigator = {
-    enable = mkEnableOption "Tmux-Navigator (see https://github.com/christoomey/vim-tmux-navigator for tmux installation instruction)";
-
-    package = helpers.mkPackageOption "tmux-navigator" pkgs.vimPlugins.tmux-navigator;
-
-    tmuxNavigatorSaveOnSwitch = helpers.mkNullOrOption (lib.types.enum [1 2]) ''
-      null: don't save on switch (default value)
-      1: update (write the current buffer, but only if changed)
-      2: wall (write all buffers)
-    '';
-
-    tmuxNavigatorDisableWhenZoomed = helpers.mkNullOrOption (lib.types.enum [1]) ''
-      null: unzoom when moving from Vim to another pane (default value)
-      1: If the tmux window is zoomed, keep it zoomed when moving from Vim to another pane
-    '';
-
-    tmuxNavigatorNoWrap = helpers.mkNullOrOption (lib.types.enum [1]) ''
-      null: move past the edge of the screen, tmux/vim will wrap around to the opposite side (default value)
-      1: disable wrap
-    '';
-  };
-
-  config = mkIf cfg.enable {
-    extraPlugins = [cfg.package];
-
-    globals = {
-      tmux_navigator_save_on_switch = cfg.tmuxNavigatorSaveOnSwitch;
-      tmux_navigator_disable_when_zoomed = cfg.tmuxNavigatorDisableWhenZoomed;
-      tmux_navigator_no_wrap = cfg.tmuxNavigatorNoWrap;
+  # TODO: Introduced 2024-03-19, remove on 2024-05-19
+  deprecations = let
+    pluginPath = ["plugins" "tmux-navigator"];
+    option = s: pluginPath ++ [s];
+    setting = s: pluginPath ++ ["settings" s];
+    settingStr = s: concatStringsSep "." (setting s);
+  in [
+    (
+      mkRenamedOptionModule
+      (option "tmuxNavigatorSaveOnSwitch")
+      (setting "save_on_switch")
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorDisableWhenZoomed")
+      "Use `${settingStr "disable_when_zoomed"}` option."
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorNoWrap")
+      "Use `${settingStr "no_wrap"}` option."
+    )
+  ];
+in
+  helpers.vim-plugin.mkVimPlugin config {
+    # FIXME document tmux-side installation instructions
```

Comment:
I believe `mkVimPlugin` would need a "`extraEnableDesc`" parameter or similar, that way the default `"Whether to enable $name}."` description could be joined with extra documentation.

In this case something along the lines of `extraEnableDesc = "Note: tmux must be configured separately for tmux-navigator to work correctly. See [upstream docs](...) for instructions."`

I wonder if it's also worth having a small user-guide section covering this in more detail?

Reply:
Good idea.
I made a PR: https://github.com/nix-community/nixvim/pull/1297

Reply:
Rebased on main to make use of it :+1:

Reply:
Hm, getting newline errors...

```
$ nix build #docs --show-trace
error: builder for '/nix/store/2yc247fwdjc66324wgrrr421cvmnb4lc-nixvim-docs.drv' failed with exit code 2;
       last 2 log lines:
       > Running phase: buildPhase
       > /nix/store/3q4aagwsn8dp474143v2ixzqjrqibpv1-copy_docs: command substitution: line 3439: syntax error near unexpected token `newline'
```

Removing line breaks from the description allows docs to build... Are newlines supposed to work?